exercise_2/src/exercise_2.cpp

```diff
@@ -0,0 +1,99 @@
+#include <cstdio>
+#include <climits>
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include<unordered_set>
+
+using namespace std;
+
+int const ROOT = {0};
+
+struct Edge {
+	int from_vertice = {0};
+	int to_vertice = {0};
+	int weight = {0};
+};
+
+class Graph
+{
+	public:
+		Graph(vector<Edge> edges, int vertices);
+		~Graph() = default;
+		void setShortestPathes(void);
+		void printShortestPathes(void);
+	private:
+		vector< vector<int> > graph_matrix;
+		int size;
+		vector<int> distance;
+		int minDistance(vector<int> distance, bool isVisited[]);
+};
+
+Graph::Graph(vector<Edge> edges, int vertices)
+{
+	size = vertices;
+	graph_matrix.resize(size, vector<int> (size));
+	for (Edge edge: edges)
+		{
+		    graph_matrix[edge.from_vertice][edge.to_vertice] = edge.weight;
+			graph_matrix[edge.to_vertice][edge.from_vertice] = edge.weight;
+		};
+};
+
+int Graph::minDistance(vector<int> distance, bool isVisited[])
+{
+    int min = INT_MAX, min_index = 0;
+    for (int v = 0; v < size; ++v)
+        if (isVisited[v] == false && distance[v] <= min)
+            min = distance[v], min_index = v;
+    return min_index;
+};
+
+void Graph::setShortestPathes()
```

Comment:
If a method is called `set` it should take some arguments and set something in the object.
If a method is called `get` is should usually take no arguments and return something.
In this case you could use some method like `.computePatches()` for example

Btw. free functions in C++ are just called functions, while functions defined in a class are object `methods`

Reply:
Renamed method